index.html

```diff
@@ -2866,32 +2850,52 @@ <h3><dfn>Writing or pushing content</dfn></h3>
         <a href="https://www.w3.org/2001/tag/doc/URNsAndRegistries-50.html">
         W3C TAG recommendations</a> to avoid using URNs in the Web.
         Therefore, Web NFC does not use the URN prefix neither when reading
-        or writing <a>NDEF messages</a>, but requires the leading colon ('`:`')
-        before the type name.
+        or writing <a>NDEF messages</a>.
       </p>
       <p>
-        To <dfn>validate external type</dfn> given a |input:USVString| run
+        To <dfn>validate external type</dfn> given |input:USVString|, run
         these steps:
       </p>
       <ol class=algorithm id="validate-external-type">
         <li>
           If |input:string| is not a {{USVString}} or it is empty, or its
-          length exceeds 255 bytes, or it contains white spaces,
+          length exceeds 255 bytes, return `false`.
+        </li>
+        <li>
+          Find the last occurrence of colon (`':'`) in |input|. If not found,
           return `false`.
         </li>
         <li>
-          If |input| does not start with colon ('`:`'), return |input|.
-          <p class="issue">
-            Validating domain name needs an algorithm.
-          </p>
+          Otherwise let |domain| be the part of |input| preceding the last colon,
+          and |type| be the part following the last colon.
+        </li>
+        <li>
+          Run the following sub-steps of <a href="">host parsing</a> on |domain|:
+          <ul>
+            <li>
+              Let |asciiDomain| be the result of running
+              <a href="https://url.spec.whatwg.org/#concept-domain-to-ascii">
+              domain to ASCII</a> on |domain|.
+            </li>
+            <li>
+              If |asciiDomain| is failure, return `false`.
+            </li>
+            <li>
+              If |asciiDomain| contains a
+              <a href="https://url.spec.whatwg.org/#forbidden-host-code-point">
+              forbidden host code point</a>, return `false`.
+            </li>
+          </ul>
+        </li>
+        <li>
+          If |type| is empty, return `false`.
         </li>
         <li>
-          If |input| starts with colon ('`:`'), let |domain:string| be the
-          <a data-cite="url#concept-url-serializer">serialization</a> of the
-          <a data-cite="html">effective domain</a> of the {{Document/origin}}.
+          If |type| does not match the `custom-type` definition from the
+          <a>external type name ABNF</a>, return `false`.
```

Comment:
could we improve this text? "match ABNF" stinks.

Reply:
Yeah, its basically ASCII without whitespaces and a few extra special chars. You can write an algorithm.

- Is ASCII
- has whitespaces
- Contains other non-valid characters

Reply:
And no control characters

Reply:
Actually they accept a very weird selection!

0x24 $
0x27 '
0x28 (
0x29 )
0x2A *
0x2B +
0x2C ,
0x2D -
0x2E .
0x3B ;
0x3D =
0x40 @ 
0x5F _

Reply:
URL spec uses wording like 

> Otherwise, if byte is 0x25 (%) and the next two bytes after byte in input are not in the ranges 0x30 (0) to 0x39 (9), 0x41 (A) to 0x46 (F), and 0x61 (a) to 0x66 (f), all inclusive, append byte to output.

Reply:
and

> The URL code points are ASCII alphanumeric, U+0021 (!), U+0024 ($), U+0026 (&), U+0027 ('), U+0028 LEFT PARENTHESIS, U+0029 RIGHT PARENTHESIS, U+002A (*), U+002B (+), U+002C (,), U+002D (-), U+002E (.), U+002F (/), U+003A (:), U+003B (;), U+003D (=), U+003F (?), U+0040 (@), U+005F (_), U+007E (~), and code points in the range U+00A0 to U+10FFFD, inclusive, excluding surrogates and noncharacters.

Reply:
The type code points are ASCII alphanumeric, U+0024 ($), U+0027 ('), U+0028 LEFT PARENTHESIS, U+0029 RIGHT PARENTHESIS, U+002A (*), U+002B (+), U+002C (,), U+002D (-), U+002E (.), U+003B (;), U+003D (=), U+0040 (@), U+005F (_).

Reply:
![image](https://user-images.githubusercontent.com/1101339/70894082-99205f80-1fec-11ea-8f08-b789b6efcd72.png)

Reply:
I have updated (amended) the steps, please check again.